src/lib/session/SessionManager.ts

```diff
@@ -94,29 +120,74 @@ export class SessionManager {
     newAccount: string,
     { targetWorkspace = 'master', authMethod = 'oauth', useCachedToken = true }: LoginOptions
   ) {
-    const currentToken = new Token(this.sessionPersister.getAccountToken(newAccount))
-    if (useCachedToken && currentToken.isValid()) {
-      this.currAccount = newAccount
-      this.currWorkspace = targetWorkspace
-      this.currToken = currentToken
-      this.saveCredentials()
+    if (this.account !== newAccount) {
+      this.state.lastAccount = this.account
+      this.state.lastWorkspace = null
+    }
+
+    const cachedToken = new Token(this.sessionPersister.getAccountToken(newAccount))
+    if (useCachedToken && cachedToken.isValid()) {
+      this.state.account = newAccount
+      this.state.workspace = targetWorkspace
+      this.state.tokenObj = cachedToken
+      this.flushState()
```

Comment:
This name `flushState` gives the impression we are disposing the state, but we are actually refreshing it.

Reply:
How about `saveState`? The idea of `flushState` was to send it to the underlying sessionPersister implementation to save it to disk, for example

Reply:
I think this one works better. `saveState` or `updateState`.